### PR TITLE
Gh 358 Upgrade to webpack 4

### PR DESCRIPTION
**Pull request by VassilIordanov**
NCI-Agency/anet#358

### CODE REVIEW

File: client/config/webpack.common.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    }
+                ]
+            }
+        ]
+    },
+    plugins: [
+        new InterpolateHtmlPlugin({PUBLIC_URL: publicUrl}),
+        new HtmlWebpackPlugin({inject: true, template: paths.appHtml})
+    ]
+};
```
Comment (gjvoosten) [inline client/config/webpack.common.js]: Add newline at EOF.

File: client/config/webpack.prod.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    filename: 'static/js/[name].[chunkhash:8].js',
+    chunkFilename: 'static/js/[name].[chunkhash:8].chunk.js',
+},
+  plugins: [
+    new UglifyJSPlugin({
+      parallel: true,
+      cache: true
+    })
+  ]
+});
```
Comment (gjvoosten) [inline client/config/webpack.prod.js]: Add newline at EOF.

**Comment by VassilIordanov** [comment]
btw, I recommend using nodejs >=8 with this one

File: client/src/pages/people/Form.js [hunk 3]
```diff
@@ -363,11 +366,17 @@ export default class PersonForm extends ValidatableFormWrapper {
 				case 'leftVacant':
 				case 'hasReplacement':
 					// reset account?
-					if (confirm('Are you sure you want to reset this account?')) {
-						let { person } = this.state
-						person.status = 'INACTIVE'
-						this.updatePerson(person, true, optionValue === 'needNewAccount')
-					}
+					confirmAlert({
```
Comment (VassilIordanov) [inline client/src/pages/people/Form.js]: @gjvoosten please review. Any idea on how to maintain an consistent style?

Comment (gjvoosten) [reply client/src/pages/people/Form.js]: @vassil Please note: if a super user resets an account, the `confirmLabel` is not correct (it's only valid for the `needNewAccount` case).

File: client/src/pages/people/Form.js [hunk 4]
```diff
@@ -21,6 +21,9 @@ import utils from 'utils'
 import CALENDAR_ICON from 'resources/calendar.png'
 import '../../components/NameInput.css'
 
+import { confirmAlert } from 'react-confirm-alert'; 
+import 'react-confirm-alert/src/react-confirm-alert.css'
```
Comment (gjvoosten) [inline client/src/pages/people/Form.js]: Trailing semicolon and trailing whitespace. ☺

File: client/src/pages/people/Form.js (hunk 3 above)
Comment (gjvoosten) [reply client/src/pages/people/Form.js]: @vassil I've updated the `confirmLabel`. As to styling: the defaults are not so bad. We could make the button backgrounds blue if you wish. Too bad there's no different style classes for the cancel and confirm buttons, so having the cancel one black text on a white background and the confirm one white text on a blue background would have to depend on the order of the buttons.

**Pull request merged**